### PR TITLE
Update main.nf to set TMPDIR from environment

### PR DESCRIPTION
**Pull request by anilthanki**
once merged TMPDIR from env.sh should be removed.

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -63,6 +63,11 @@ process download_fastqs {
         set val(runId), file("${cdnaFastqFile}"), file("${barcodesFastqFile}"), val(barcodeLength), val(umiLength), val(end), val(cellCount) into DOWNLOADED_FASTQS
 
     """
+        if ! [ -z "$ATLAS_TMPDIR" ]; then 
+            TMPDIR=$ATLAS_TMPDIR; 
+        # else 
+        #    echo "TMPDIR is unset";
```
Comment (pmb59) [inline main.nf]: if unnecessary, I would remove. 
Alternatively:
```suggestion
        else 
                echo "NOTE: ATLAS_TMPDIR not defined"
```

**Review by pmb59 — COMMENTED** [review]
You mention in the description of the PR that

> once merged TMPDIR from env.sh should be removed.

If env.sh is in a GitHub repo, perhaps you want to open a PR there, remove TMPDIR, and add a link here so in the future it is clear that both PRs should be merge simultaneously.

**Comment by anilthanki** [comment]
> You mention in the description of the PR that
> 
> > once merged TMPDIR from env.sh should be removed.
> 
> If env.sh is in a GitHub repo, perhaps you want to open a PR there, remove TMPDIR, and add a link here so in the future it is clear that both PRs should be merge simultaneously.

It's an internal environment file, I dont think that env.sh is on GitHub

**Comment by anilthanki** [comment]
Tested on scxa test environment and its not failing at fastq-provider step..

temp fastq-provider files created at `$ATLAS_TEMP`

File: main.nf [hunk 2]
```diff
@@ -63,6 +63,11 @@ process download_fastqs {
         set val(runId), file("${cdnaFastqFile}"), file("${barcodesFastqFile}"), val(barcodeLength), val(umiLength), val(end), val(cellCount) into DOWNLOADED_FASTQS
 
     """
+        if ! [ -z "$ATLAS_TMPDIR" ]; then 
+            TMPDIR=$ATLAS_TMPDIR; 
+        else 
+                echo "NOTE: ATLAS_TMPDIR not defined"
```
Comment (pmb59) [inline main.nf]:
```suggestion
            echo "NOTE: ATLAS_TMPDIR not defined"
```
indent correction.

**Review by pmb59 — APPROVED** [review]
@anilthanki  I'll approve as you said it's working as expected.

**Comment by anilthanki** [comment]
> @anilthanki I'll approve as you said it's working as expected.

Thanks

**Pull request merged**